x-pack/legacy/plugins/ml/server/routes/filters.ts

```diff
@@ -0,0 +1,227 @@
+  return mgr.deleteFilter(filterId);
+}
+
+export function filtersRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters Gets filters
+   * @apiName GetFilters
+   * @apiDescription Retrieves the list of filters which are used for custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filters list of filters
+   */
+  router.get(
+    {
+      path: '/api/ml/filters',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getAllFilters(context);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {get} /api/ml/filters/:filterId Gets filter by ID
+   * @apiName GetFilterById
+   * @apiDescription Retrieves the filter with the specified ID.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter the filter with the specified ID
+   */
+  router.get(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await getFilter(context, request.params.filterId);
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {put} /api/ml/filters Creates a filter
+   * @apiName CreateFilter
+   * @apiDescription Instantiates a filter, for use by custom rules in anomaly detection.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter created filter
+   */
+  router.put(
+    {
+      path: '/api/ml/filters',
+      validate: {
+        body: schema.object({ ...createFilterSchema }),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const body = request.body;
+        const resp = await newFilter(context, body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup Filters
+   *
+   * @api {put} /api/ml/filters/:filterId Updates a filter
+   * @apiName UpdateFilter
+   * @apiDescription Updates the  description of a filter, adds items or removes items.
+   *
+   * @apiSuccess {Boolean} success
+   * @apiSuccess {Object} filter updated filter
+   */
+  router.put(
+    {
+      path: '/api/ml/filters/{filterId}',
+      validate: {
+        params: schema.object({ filterId: schema.string() }),
+        body: schema.object({ ...updateFilterSchema }),
```

Comment:
Same as above

Reply:
Done in https://github.com/elastic/kibana/pull/57597/commits/8cd2ec85d391ca81e15f50b70a9531e01891fe68